util/src/main/java/io/kubernetes/client/util/ConfigBuilder.java

```diff
@@ -0,0 +1,272 @@
+
+	private static final Logger log = Logger.getLogger(Config.class);
+
+	public String getUserName() {
+		return userName;
+	}
+
+	public ConfigBuilder setUserName(String userName) {
+		this.userName = userName;
+		return this;
+	}
+
+	public String getPassword() {
+		return password;
+	}
+
+	public ConfigBuilder setPassword(String password) {
+		this.password = password;
+		return this;
+	}
+
+	public String getApiKey() {
+		return apiKey;
+	}
+
+	public ConfigBuilder setApiKey(String apiKey) {
+		this.apiKey = apiKey;
+		return this;
+	}
+
+	public String getBasePath() {
+		return basePath;
+	}
+
+	public ConfigBuilder setBasePath(String basePath) {
+		this.basePath = basePath;
+		return this;
+	}
+
+	public File getCertificateAuthorityFile() {
+		return certificateAuthorityFile;
+	}
+
+	public ConfigBuilder setCertificateAuthority(File certificateAuthorityFile) {
+		this.certificateAuthorityFile = certificateAuthorityFile;
+		this.verifyingSsl = true;
+		return this;
+	}
+
+	public String getCertificateAuthorityData() {
+		return certificateAuthorityData;
+	}
+
+	public ConfigBuilder setCertificateAuthority(String certificateAuthorityData) {
+		this.certificateAuthorityData = certificateAuthorityData;
+		this.verifyingSsl = true;
+		return this;
+	}
+
+	public ConfigBuilder setClusterMode() {
+		this.clusterMode = true;
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(KubeConfig config) {
+		this.kubeConfig = config;
+		return this;
+	}
+
+	public ConfigBuilder setDefaultKubeConfigMode() {
+		this.defaultKubeConfigMode = true;
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(String fileName) throws FileNotFoundException {
+		this.kubeConfig = KubeConfig.loadKubeConfig(new FileReader(fileName));
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(Reader input) {
+		this.kubeConfig = KubeConfig.loadKubeConfig(input);
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(InputStream stream) {
+		this.kubeConfig = KubeConfig.loadKubeConfig(new InputStreamReader(stream));
+		return this;
+	}
+
+	public KeyManager[] getKeyMgrs() {
+		return keyMgrs;
+	}
+
+	public ConfigBuilder setKeyMgrs(KeyManager[] keyMgrs) {
+		this.keyMgrs = keyMgrs;
+		return this;
+	}
+
+	public boolean isVerifyingSsl() {
+		return verifyingSsl;
+	}
+
+	public ConfigBuilder setVerifyingSsl(boolean verifyingSsl) {
+		this.verifyingSsl = verifyingSsl;
+		return this;
+	}
+
+	public boolean isDefaultClientMode() {
+		return defaultClientMode;
+	}
+
+	public ConfigBuilder setDefaultClientMode() {
+		this.defaultClientMode = true;
+		return this;
+	}
+
+	public String getApiKeyPrefix() {
+		return apiKeyPrefix;
+	}
+
+	public ConfigBuilder setApiKeyPrefix(String apiKeyPrefix) {
+		this.apiKeyPrefix = apiKeyPrefix;
+		return this;
+	}
+
+	public ApiClient build() {
+		ApiClient client = new ApiClient();
+
+		if( kubeConfig !=null) {
+			client = Config.fromConfig(kubeConfig);
+		}
+
+		if(defaultKubeConfigMode == true) {
+			try {
```

Comment:
Let's get rid of the try/catch and just have `build()` throw, I think that's better.

Reply:
Better: move `KubeConfig.loadDefaultKubeConfig()` up into the `setDefaultMode()` function.

Reply:
@brendandburns 
is the following what you intend

```java
public ConfigBuilder setDefaultMode() {
		this.defaultMode = true;
		return this;
	}
build()  throws Exception{
....
if(defaultMode == true) {
	client = Config.fromConfig(KubeConfig.loadDefaultKubeConfig());
}
...
}
```

Reply:
@kondapally1989 no, what I was thinking was this:

```java
public ConfigBuilder setDefaultMode() throws IOException {
      this.client = config.fromConfig(KubeConfig.loadDefaultKubeConfig());
}
```

Reply:
@brendandburns 
 if setDefaultMode()  points to Config.fromConfig(KubeConfig.loadDefaultKubeConfig())

and setDefaultClientMode points to Config.defaultClient();

isn't the naming convention confusing. instead of setDefaultMode() cant we have setDefaultKubeConfigMode()